### PR TITLE
bpf: Improve compilation coverage

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request improves compilation coverage for bpf_netdev and bpf_lxc and fixes new warnings. See commit messages for details.

### CODE REVIEW

**Review by tklauser — APPROVED** [review]

**Comment by pchaigno** [comment]
test-me-please

**Review by qmonnet — APPROVED** [review]

**Comment by pchaigno** [comment]
test-me-please

**Comment by qmonnet** [comment]
> Coverage decreased (-0.005%) to 45.505% when pulling **[f381335](https://github.com/cilium/cilium/commit/f381335a6c0e36f0aec9bc25086e3c42a95b99d8) on pr/pchaigno/improve-bpf-compilation-coverage** into **[7c023b5](https://github.com/cilium/cilium/commit/7c023b53832de6c6e6fb825d5c678925e34b27ef) on master**.

Coverage bot disagrees with your PR subject though :p

**Comment by pchaigno** [comment]
@tklauser I fixed one more warning since you reviewed: https://github.com/cilium/cilium/pull/10712/files#diff-ee19aada4747b2de1058a11c26271729R610.

**Comment by tklauser** [comment]
> @tklauser FYI, I fixed one more warning since you reviewed: https://github.com/cilium/cilium/pull/10712/files#diff-ee19aada4747b2de1058a11c26271729R610.

Thanks, still LGTM.

**Review by christarazi — APPROVED** [review]

File: bpf/Makefile
Comment (brb) [inline bpf/Makefile]: If you add `ENABLE_DSR_HYBRID` to the list, this should improve the coverage even more.

Also, you could add `ENABLE_HOSTPORT` which was recently introduced in #10592.

Comment (pchaigno) [reply bpf/Makefile]: Thanks! Done.

**Review by brb — APPROVED** [review]
Thanks!

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
Good news/bad news: thanks to the increased coverage of this pull request, the tests are now failing because of the padding issue introduced in #10626 and fixed in #10728. I'll rebase once #10728 is merged.

**Comment by pchaigno** [comment]
test-me-please

**Comment by aanm** [comment]
@pchaigno it seems a real failure:

```
Exitcode: 1 
Stdout:
 	 => Loading bpf_hostdev_ingress.c:to-host...
	 => Loading bpf_ipsec.c:from-netdev...
	 => Loading bpf_lxc.c:from-container...
	 8 bytes struct bpf_elf_map fixup performed due to size mismatch!
	 
Stderr:
 	 Map 'test_cilium_events' loaded as pinned!
	 Map 'test_cilium_lxc' loaded as pinned!
	 Map 'test_cilium_metrics' loaded as pinned!
	 Map 'test_cilium_policy' loaded as pinned!
	 Map 'test_cilium_policy_65535' loaded as pinned!
	 Map 'test_cilium_calls_65535' loaded as pinned!
	 Map 'test_cilium_tunnel_map' loaded as pinned!
	 Map 'test_cilium_ipcache' loaded as pinned!
	 Map 'test_cilium_encrypt_state' loaded as pinned!
	 Map 'test_cilium_lb6_reverse_nat' loaded as pinned!
	 Map 'test_cilium_lb6_services' loaded as pinned!
	 Map 'test_cilium_lb6_backends' loaded as pinned!
	 Map 'test_cilium_lb4_reverse_nat' loaded as pinned!
	 Map 'test_cilium_lb4_services' loaded as pinned!
	 Map 'test_cilium_lb4_backends' loaded as pinned!
	 Map 'test_cilium_signals' loaded as pinned!
	 Map 'test_cilium_ct_tcp6_65535' loaded as pinned!
	 Map 'test_cilium_ct_any6_65535' loaded as pinned!
	 Map 'test_cilium_ct_tcp4_65535' loaded as pinned!
	 Map 'test_cilium_ct_any4_65535' loaded as pinned!
	 Map 'cilium_snat_v4_external' loaded as pinned!
	 Map 'cilium_snat_v6_external' loaded as pinned!
	 Map object 'test_cilium_neigh4' rejected: Invalid argument (22)!
	  - Type:         9
	  - Identifier:   0
	  - Pinning:      2
	  - Size key:     4
	  - Size value:   8
	  - Max elems:    524288
	  - Flags:        0
	 
	 Error loading maps into kernel!
	 Error fetching ELF ancillary data!
	 Unable to load program
	 
```

**Comment by aanm** [comment]
test-me-please

**Comment by pchaigno** [comment]
@aanm It is a real failure. I'm working on it. It's a bit more involved than I thought :confused: 
I'll send an update in a bit.

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
@joestringer @brb I can't enable NodePort in the verifier complexity test (RuntimeVerifier) because it runs on 4.9. I'm not sure how to proceed here: we likely want to continue testing on 4.9, but should also test the latest additions. Should I simply define a new make target to compile object files with NodePort & co. for when we want to check complexity manually with net-next?

**Comment by brb** [comment]
@pchaigno How much work would it require to move `runtime/verifier.go` to `k8sT/verifier.go` and make it part of the k8sT test suite?

**Comment by pchaigno** [comment]
@brb From a quick look at the dependencies, it should be relatively straightforward. Is your idea to move it to `k8sT/` then test each set of options depending on 4.9 vs. net-next?

**Comment by brb** [comment]
> Is your idea to move it to k8sT/ then test each set of options depending on 4.9 vs. net-next?
Yep.

**Comment by pchaigno** [comment]
> From a quick look at the dependencies, it should be relatively straightforward. 

@brb Scratch that. We would need new helpers to properly stop and cleanup Cilium state. It looks like there is some code to do that in K8sUpdates, but I'd rather leave that to a subsequent pull request.

There is no regression in not compile testing NodePort & co. here since they were already not compile tested before (because `FROM_HOST` was always set).

**Comment by pchaigno** [comment]
test-me-please

**Comment by joestringer** [comment]
Please do follow up on the runtime complexity test, either by:
* Defining a new target with max complexity for newest kernels, or
* Detecting host support and compiling based on that, or
* Shifting the verifier test into the k8s tests (probably also requires detecting host support?)

**Pull request merged**
